src/Symfony/Component/Finder/Iterator/SortableIterator.php

```diff
@@ -70,9 +74,11 @@ public function __construct(\Traversable $iterator, $sort, bool $reverseOrder =
                 return $order * ($a->getMTime() - $b->getMTime());
             };
         } elseif (self::SORT_BY_NONE === $sort) {
-            $this->sort = $order;
+            $this->sort = null;
```

Comment:
This does not preserve the order, that was stored before.

Reply:
Right. I was probably a bit too eager here.